src/core/coap/coap_secure.cpp

```diff
@@ -320,6 +320,11 @@ CoapSecure &CoapSecure::GetOwner(const Context &aContext)
     return coap;
 }
 
+void CoapSecure::HandleRetransmissionTimer(Timer &aTimer)
+{
+    GetOwner(aTimer).Coap::HandleRetransmissionTimer();
```

Comment:
Talked to @jwhui offline and he explained the issue. Basically we have two instances of the `Coap`. The current approach addresses it and is fine...